client/src/main/java/io/pravega/client/admin/ReaderGroupManager.java

```diff
@@ -73,11 +72,10 @@ public static ReaderGroupManager withScope(String scope, ClientConfig clientConf
      * may block.
      * @param groupName The name of the group to be created.
      * @param config The configuration for the new ReaderGroup.
-     * @return True if ReaderGroup was created.
-     * @throws ConfigMismatchException If the reader group already exists with a different configuration. Use {@link ReaderGroup#resetReaderGroup} to change
-     * the reader group configuration.
+     * @return The {@link ReaderGroupConfig} of the ReaderGroup. If a ReaderGroup already exists then the older configuration
+     * is returned.
      */
-    boolean createReaderGroup(String groupName, ReaderGroupConfig config) throws ConfigMismatchException;
+    ReaderGroupConfig createReaderGroup(String groupName, ReaderGroupConfig config);
```

Comment:
This is replacing the configuration unconditionally, right? As we spoke offline, the concern is that the application accidentally overwrites an existing configuration, so is the idea that we pass the responsibility of checking the configuration to the application? By the time the application is able to check, it might be late.

If you can elaborate on how you see the application using it, I'd appreciate it.

Reply:
The API does not overwrite the ReaderGroup configuration. If the ReaderGroup(RG) exists then the current configuration of the RG is returned.

Reply:
It would help to merge this API and the getReaderGroup() API into a single - getOrCreateReaderGroup() API that returns a ReaderGroup. Also ReaderGroupConfiguration should be part of the ReaderGroup object.
I don't see the point in having 2 separate APIs here, if create is going to return the same existing Reader Group configuration.... its actually doing a getOrCreate() on the ReaderGroup configuration without explicitly saying so....

From API design perspective, it looks strange that a **createReaderGroup()** API returns **ReaderGroupConfiguration**. I would expect it to either return nothing (_if no exception is thrown the reader group is created_) or return a reference to the created reader group (which essentially is doing a getOrCreateReaderGroup())

Reply:
For the proposal for `getOrCreateReaderGroup`, it's a better and more natural solution to me. It can make the current Flink connector usage simpler into one line.
For the responsibility to close the reader group, I think it should leave to the user and it's always in this way.
Deleting the old `createReaderGroup` call may break the compatibility, we need to check and test if we choose this, and I would propose deprecate it now and then delete in the future releases.

Reply:
@shrids Doesn't this break backward compatibility? How would we handle existing users that have already written code against this API?

Reply:
The change to return a boolean was done in the 0.10.x/master branch. The API in 0.9.x is 
```
void createReaderGroup(String groupName, ReaderGroupConfig config);
```
Hence this does not break backward compatibility.

The drawback of the API in 0.9.x is that if the reader group exists it is not initmated to the user. It also does not indicate to the user if reader group config passed is different from that of the existing reader group configuration.

Reply:
I have added an helper API ` ReaderGroup getOrCreate(String groupName, ReaderGroupConfig config)` in the ReaderGroupManager.

Reply:
in my opinion it is better to save users from breaking changes in the API

can't we keep `boolean` as return type ?